File: tests/go/cases/case_211_shadowed_sink_local_01.go
package escape_tests

var retainedCase211 = []map[string]string{}

func Case211ShadowedSinkLocal01(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "shadowed_sink_local_01",
		"entity": "extreme",
		"stage": "stress",
		"input": raw,
		"checksum": "shadowed_sink_local_01:" + raw,
	}
	retainedCase := []map[string]string{}
	retainedCase = append(retainedCase, payload)
	// SAFE: local shadow variable only.
	return payload["checksum"]
}

File: tests/rust/cases/case_211_shadowed_sink_local_01.rs
use std::collections::HashMap;
use std::sync::{Mutex, OnceLock};

static RETAINED_AUDIT: OnceLock<Mutex<Vec<HashMap<String, String>>>> = OnceLock::new();
static RETAINED_HANDLERS: OnceLock<Mutex<Vec<String>>> = OnceLock::new();

pub fn case_211_shadowed_sink_local_01(input: String) -> String {
    let task_name = "shadowed_sink_local_01".to_string();
    let raw = if input.is_empty() { "sample".to_string() } else { input };
    let mut payload: HashMap<String, String> = HashMap::new();
    payload.insert("task".to_string(), task_name.clone());
    payload.insert("entity".to_string(), "extreme".to_string());
    payload.insert("stage".to_string(), "stress".to_string());
    payload.insert("input".to_string(), raw.clone());
    payload.insert("checksum".to_string(), format!("{}:{}", task_name, raw.len()));
    let mut retained_audit: Vec<HashMap<String, String>> = Vec::new();
    retained_audit.push(payload.clone());
    // SAFE: local shadow does not touch global sink.
    payload.get("checksum").cloned().unwrap_or_default()
}
